core/protocol.py

```diff
@@ -62,3 +63,29 @@ def is_registered():
     """
     address = get_backend_address()
     return address == current_app.config['PUBLIC_KEY']
+
+def get_delivery(delivery_hash):
+    """
+    Returns owner, bytes_requested, and bytes_delivered for a delivery
+    """
+    d = get_datatrust()
+    delivery = call(d.get_delivery(delivery_hash))
+    return delivery
+
+def listing_accessed(delivery_hash, listing, amount):
+    """
+    Commit to protocol the listing accessed details
+    """
+    d = get_datatrust()
+    tx = send_or_transact(d.listing_accessed(listing, delivery_hash, amount))
+    # TODO: seems like we should be waiting here for the transaction to be mined before delivering content to the buyer
```

Comment:
i don't think we need to block here. the two aren't coupled and if that call errors, it's not the user's fault.

Reply:
Updated